### PR TITLE
[PD-3027][PD-3163] Site traffic report.

### PR DESCRIPTION
**Pull request by almathew**
## What's this PR do?
Adds the Site Traffic report

## Screenshots (if appropriate)
site-traffic-report

## Where should the reviewer start?
There is a migration to add the report. Migrations are pretty out of hand this sprint...
Most changes were on the analytics API/utils side.
Minimal changes were done to get the analytics_event datasource working.

## How should this be manually tested?
Run the migrations.
Run `python manage.py make_data`.
Navigate to the report intentions page - site traffic should be an option.
Click through to the site traffic report and mess around. Stuff should work.

## Any background context you want to provide?
CSS for the table is still messed up for this PR and the next one I'm going to put up. 
I'm going to bother @mwarnsleyDE about what he thinks we should do for that. So a fix for that is coming separately.

### CODE REVIEW

File: analytics/urls.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
 view_patterns = patterns(
     'analytics.views',
-    url(r'^main/(?P<analytics_type>[\w-]+)$', 'analytics_main', name='analytics_main'),
-    url(r'^main$', 'analytics_main', name='analytics_main'),
+    url(r'^main/(?P<analytics_type>[\w-]+)/$', 'analytics_main', name='analytics_main'),
+    url(r'^main$/', 'analytics_main', name='analytics_main'),
```
Comment (tdphillips) [inline analytics/urls.py]: I think these two characters might have gotten transposed.

Comment (almathew) [inline analytics/urls.py]: Good catch. Fixed.

**Comment by tdphillips** [comment]
I ran the migrations and then `make_data`. This resulted in some weird results.

I cleared all of my job views and reran `make_data`, thinking I might have had some weird job views. I ended up with basically identical results.
2017-04-21-155638_862x415_scrot

**Comment by almathew** [comment]
@mwarnsleyDE I'm not seeing what @tdphillips but I am seeing some wonkiness. I'm going to work on this on my own this weekend for sure, but if you have any insight into what might be going on that'd be great :+1:

**Comment by almathew** [comment]
@tdphillips what url was that specifically? I thought @mwarnsleyDE had found all the areas where "Job Views" was hardcoded into the legend.

**Comment by tdphillips** [comment]
@almathew: `/analytics/view/main/site-traffic/`

**Comment by almathew** [comment]
@mwarnsleyDE some more info on what I'm seeing:
```Warning: Failed propType: Invalid prop `metricTotalThisPage` of type `string` supplied to `Table`, expected `number`. Check the render method of `TableContainer`.```

```Warning: Failed propType: Invalid prop `data[0].value` supplied to `Bar`. Check the render method of `BarChart`.```

And my personal favorite, this masterpiece
screen shot 2017-04-21 at 4 05 04 pm

**Comment by tdphillips** [comment]
Probably just another "me" issue but this is what I get after choosing a country and highlighting a state.
2017-04-21-160911_389x258_scrot

**Comment by almathew** [comment]
There might be a slight ordering issue for the report dimensions, or this might just be me. Can someone run all the migrations and double check?
screen shot 2017-04-23 at 11 25 16 am

**Comment by almathew** [comment]
Ok, all the JavaScript is fixed. 
@mwarnsleyDE turns out the ordering thing was becoming an issue, I did this by doing what we had discussed and having the Python side let the JS know what is broken. You might take a look at what I did because, while I tested everything and it seems to work, we all know what my JS skills are like.

I've "fixed" the CSS issue by removing blank columns from the footer. Now the only columns in the footer are ones that actually have data in them. I'm not certain this is a good long-term solution, but it's good enough for DEAM.
screen shot 2017-04-23 at 11 31 36 am

At this point I should've addressed all the concerns that @tdphillips and I have raised by making the changes to how the primary key/label was determined. 

So the only remaining issue is the primary dimensions specified for the "Job Locations" site analytics report. If a reviewer could double check that to confirm that it's just my database being wrong, that would be great :+1:

**Pull request merged**